src/redux/categories/categories.sagas.js

```diff
@@ -1,31 +1,41 @@
 import { takeEvery, call, put } from 'redux-saga/effects';
-import { setCategories } from './categories.actions';
+import { push } from 'connected-react-router';
+import { setCategories, setCategoriesLoading } from './categories.actions';
+import { setError } from '../error/error.actions';
 import getItems from '../../utils/client';
 import { GET_CATEGORIES } from './categories.types';
 
-function* handleCategoriesLoad() {
+export function* handleCategoriesLoad() {
+  yield put(setCategoriesLoading(true));
   const query = `query {
-               getAllCategories {
-              categoryCode
-              _id
-              name {
-               value
-               lang
-              }
-              images {
-                large
-              }
-            }          
-             }`;
+                   getAllCategories {
+                     categoryCode
+                     _id
+                     name {
+                     value
+                     lang
+                     }
+                     images {
+                      large
+                   }
+                  }          
+                 }`;
 
   try {
     const categories = yield call(getItems, query);
     yield put(setCategories(categories.data.getAllCategories));
+    yield put(setCategoriesLoading(false));
   } catch (e) {
-    console.log(e);
+    yield call(handleCategoriesError, e);
   }
 }
 
+function* handleCategoriesError(e) {
```

Comment:
Why have you created a separate function for error handling if it is used only once?

Reply:
Removed